graphite/graphite.go

```diff
@@ -42,23 +42,31 @@ func (r *Request) Query(host string) (Response, error) {
 	if r.End != nil {
 		v.Add("until", fmt.Sprint(r.End.Unix()))
 	}
-	u := url.URL{
+	r.Url = url.URL{
 		Scheme:   "http",
 		Host:     host,
 		Path:     "/render/",
 		RawQuery: v.Encode(),
 	}
-	resp, err := DefaultClient.Get(u.String())
+}
+
+func (r *Request) Query(host string) (Response, error) {
+	r.setUrl(host)
```

Comment:
Why does setUrl need to be its own function? It's only called in one place, and short.